### DISCUSSION TITLE
Possible to add colour to the imported CSVs cells?

### DISCUSSION THREAD

**Discussion by G0nz0uk** [grafana/grafana-csv-datasource#40] [Q&A] [answered]
Hello,

Is there any way an imported CSV can have it cells colour changed based on a threshold?

I'm got a CVS imported that was running on a very old php site, but any cells with the word 'down' were red or cells showing less the '100' were red.

So some users are asking if it's possible lol

Thanks

**Comment by marcusolsson** [comment, marked as answer]
You can, but it's not up to the data source, but depends on the panel you're using. You can use the **Color scheme** and **Thresholds** field options in the panel editor. You can even override them for a specific field.

**Reply by G0nz0uk** [reply]
Interesting!  Yeah I imported a CSV to the local drive to work on and had the pane in table mode but could add any colour, I will have to try again.  Do you have a screenshot to go off to see this in action?

**Reply by G0nz0uk** [reply]
My panel doesn't give me the Colour scheme or threshold options it seems, which panel do I need to use?

**Reply by marcusolsson** [reply]
For the **Table** visualization

1. In the **Overrides** tab, select the field you want to color
2. Override the **Cell display mode** by setting it to **Color background**
3. Override the **Thresholds** and set it to the levels you need

![Screenshot 2021-02-28 at 07 33 15@2x](https://user-images.githubusercontent.com/8396880/109409980-4f285d00-7997-11eb-8213-8a69fb3cbbf2.png)

**Reply by G0nz0uk** [reply]
Amazing.  Is it possible to add colour to cells that have text?  I have a column with OK, Warning & Error.  I'd love to have green as OK, amber for Warning, Red for error?

**Reply by marcusolsson** [reply]
Well yes and no. AFAIK there's no way to set a color based on text, but you can use **Value mappings** to convert the text to a number, and then use the same method I mentioned above.

**Reply by G0nz0uk** [reply]
Works like a dream, thanks.

**Comment by jftuga** [comment]
An update to this with a slight variation. I have a column that can have 1 of 3 values: succeeded, failed, or aborted.

Tested on Grafana version 10.1.2

* Field with name: `A-series`
* Cell options -> Cell type: `Colored Background`
* Value Mappings:

```
     SUCCEEDED -> green
     FAILED -> red
     ABORTED -> red
```
